gflownet.py

```diff
@@ -555,14 +565,16 @@ def sample_batch(self, envs, n_samples=None, train=True, model=None):
                 actions = self.rng.integers(low=0, high=high, size=len(envs))
             t0_a_envs = time.time()
             assert len(envs) == actions.shape[0]
-            for env, action in zip(envs, actions):
-                seq, action, valid = env.step(action)
+            for env, action in zip(envs, actions): 
+                seq, action, valid = env.step(action) 
+                
                 if valid:
-                    parents, parents_a = env.parent_transitions(seq, action)
+                    parents, parents_a = env.parent_transitions(seq, action) #array ohe befre, tensor([2])
+                 
                     if train:
                         batch.append(
                             [
-                                tf([env.seq2obs()]),
+                                tf([env.seq2obs(seq)]),
```

Comment:
why?

Reply:
in seq2obs, why by default take as argument seq = None and then if seq is None : seq = self.seq : easier to take as input self.seq directly ?